In [44]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.graph_objects as go

In [45]:
men_data = pd.read_csv('men_1_14_3.csv')
men_data.drop(['Unnamed: 0'],axis=1,inplace=True)
men_data.head()

name Country  FX_mean_score  FX_std  FX_count  FX_D_mean_score  \
0     BOHENG ZHANG     CHN          14.43    0.26       8.0             6.02   
1      KAZUMA KAYA     JPN          13.84    0.49       6.0             5.70   
2  WATARU TANIGAWA     JPN          13.94    0.50       4.0             5.70   
3         CONG SHI     CHN          13.87    0.03       3.0             5.53   
4  DAIKI HASHIMOTO     JPN          14.26    0.50      10.0             5.95   

   FX_D_std  FX_E_mean_score  FX_E_std  FX_rank  ...  VT_D_mean_score  \
0      0.14             8.42      0.18      4.0  ...             5.60   
1      0.00             8.16      0.50     49.0  ...             5.60   
2      0.08             8.27      0.51     36.0  ...             5.64   
3      0.23             8.33      0.20     47.0  ...             5.60   
4      0.08             8.32      0.45     11.0  ...             5.60   

   VT_D_std  VT_E_mean_score  VT_E_std  VT_rank  all_scores_std  \
0      0.00             8.97      0.50     14.0            0.76   
1      0.00             8.80      0.17     42.0            0.38   
2      0.33             9.02      0.21      8.0            0.52   
3      0.00             8.30      0.66    190.0            0.59   
4      0.00             9.13      0.28      9.0            1.55   

   Standardized_Score_Sum  Total_Appearances  apparatus_participated  \
0                   86.56               46.0                       6   
1                   85.41               35.0                       6   
2                   85.27               30.0                       6   
3                   84.97               23.0                       6   
4                   84.66               51.0                       6   

   mean_score_of_all  
0              14.43  
1              14.27  
2              14.34  
3              14.21  
4              14.15  

[5 rows x 55 columns]

## Men's Floor Exercise

### Top 15 Athletes in FX by mean_score (3+ Apperances)

In [82]:
FX_data = men_data[men_data['FX_count']>3]
FX_data=FX_data[['name','Country','FX_mean_score','FX_std','FX_count','FX_D_mean_score','FX_D_std','FX_E_mean_score','FX_E_std']]
FX_data.sort_values(by='FX_mean_score', ascending=False).head(10)

name Country  FX_mean_score  FX_std  FX_count  \
16       RYOSUKE DOI     JPN          14.56    0.27       5.0   
6        CARLOS YULO     PHI          14.51    0.76      13.0   
0       BOHENG ZHANG     CHN          14.43    0.26       8.0   
9          PAUL JUDA     USA          14.34    0.34       6.0   
747    KAZUKI MINAMI     JPN          14.31    0.40       7.0   
8    KAKERU TANIGAWA     JPN          14.27    0.19       4.0   
36   ARTEM DOLGOPYAT     ISR          14.27    1.04      23.0   
4    DAIKI HASHIMOTO     JPN          14.26    0.50      10.0   
27       JAKE JARMAN     GBR          14.21    0.48      19.0   
538       HANSOL KIM     KOR          14.21    0.58       6.0   

     FX_D_mean_score  FX_D_std  FX_E_mean_score  FX_E_std  
16              6.20      0.00             8.40      0.24  
6               6.35      0.09             8.20      0.70  
0               6.02      0.14             8.42      0.18  
9               5.60      0.17             8.76      0.32  
747             6.36      0.13             7.99      0.41  
8               5.70      0.20             8.65      0.16  
36              6.21      0.19             8.14      0.89  
4               5.95      0.08             8.32      0.45  
27              6.33      0.40             7.91      0.56  
538             6.00      0.00             8.21      0.58

#### The best USA athelete in FX is Paul Juda, ranking 4th in mean_score. He has no obvious advantage comparing to top 3, with lower score, greater sd, meaning less stable performance. He has a low D_score but high E_score, so more like someone who takes lower risks.

### Distribution of Individual's FX Mean Score

In [47]:
fig = go.Figure(data=[go.Histogram(x=FX_data['FX_mean_score'],nbinsx=20)])
fig.update_layout(
    title='Histogram of FX mean score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### The distribution of FX's score is left skewed. Only 3% of athletes have a mean score above 14.3, showing a rather distinct group of elite atheletes on top level. The "best athlete" is Carlos Yulo from Philppines, who scored on average 15.0 in 4 appearances, with 0.21 sd.

### Distribution of Individual's SD

In [48]:
filter_data = FX_data[FX_data['FX_std']<5]
fig = go.Figure(data=[go.Histogram(x=filter_data['FX_std'],nbinsx=10)])
fig.update_layout(
    title='Histogram of FX SD',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### It's hard to distinguish athletes by SD, as only 2 have extremely low SD <0.2, but and most above that there's little distinction.

In [49]:
FX_sd_of_sd=filter_data['FX_std'].std()
FX_mean_sd=filter_data['FX_std'].mean()
print(FX_mean_sd.round(2),round(FX_sd_of_sd,2))

0.73 0.53


#### For FX, the mean of sd is 0.66, the sd of sd is 0.33. This shows a most atheletes show some instability in FX performances, but there is not a big enough distinction between atheletes.

### Distribution of D_Score

In [50]:
fig = go.Figure(data=[go.Histogram(x=FX_data['FX_D_mean_score'],nbinsx=20)])
fig.update_layout(
    title='Histogram of FX D_Score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### The distribution of D_Score shows a big gap at 6.1, meaning that few athletes can make a D_Score above 6.1, meaning that they perform very hard routines and have bonus points.

### Distribution of FX E_Score

In [51]:
fig = go.Figure(data=[go.Histogram(x=FX_data['FX_E_mean_score'],nbinsx=20)])
fig.update_layout(
    title='Histogram of FX E_Score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### The E_Score distribution is quite unimodal, showing that a decent amount of athletes can get a fairly high E_Score, but few get an exceptionally high E_Score (10 people above 8.4).

### Conclusion

### In general, FX relies quite heavily on D_Score to be separated from the others. Most athletes cannot perform very consistently in this apparatus. The top 3 atheletes from Philippines, China, and Japan have strong advantage and stable performance, Paul Juda from USA comes fourth and have no unique advantage, as he has a significantly lower D_score than the other 3. It is unlikely for USA to win gold medal on this appratus.

## Men's Horizontal Bar

### Top 15 Athletes in HB by mean_score (3+ Apperances)

In [83]:
HB_data = men_data[men_data['HB_count']>3]
HB_data=HB_data[['name','Country','HB_mean_score','HB_std','HB_count','HB_D_mean_score','HB_D_std','HB_E_mean_score','HB_E_std']]
HB_data.sort_values(by='HB_mean_score', ascending=False).head(10)

name Country  HB_mean_score  HB_std  HB_count  \
4    DAIKI HASHIMOTO     JPN          14.63    0.59      11.0   
0       BOHENG ZHANG     CHN          14.52    0.71       9.0   
5       BRODY MALONE     USA          14.37    0.58      13.0   
12           WEI SUN     CHN          14.31    0.48       6.0   
391      CHAOPAN LIN     CHN          14.31    0.33       5.0   
3           CONG SHI     CHN          14.31    0.76       5.0   
624      YUYA KAMOTO     JPN          14.24    0.12       5.0   
28   SHOHEI KAWAKAMI     JPN          14.23    0.35       6.0   
1        KAZUMA KAYA     JPN          14.08    0.08       6.0   
856        TIN SRBIC     CRO          13.99    0.45      20.0   

     HB_D_mean_score  HB_D_std  HB_E_mean_score  HB_E_std  
4               6.26      0.37             8.37      0.32  
0               6.18      0.38             8.34      0.40  
5               6.08      0.31             8.29      0.37  
12              6.18      0.23             8.12      0.27  
391             6.00      0.07             8.31      0.32  
3               6.10      0.28             8.21      0.54  
624             5.94      0.13             8.30      0.11  
28              6.23      0.08             8.00      0.33  
1               5.88      0.04             8.19      0.07  
856             5.70      0.21             8.30      0.40

#### The best USA athlete in HB is Brody Malone, with 14.37 mean score and 0.58 mean, 13 appearances. He is rather stable compared to the 2nd, Boheng Zhang, with a much lower sd. However, he has no apprent advantage over top 1, Daiki Hashimoto from Japan, who is higher in every score possible.

### Distribution of Individual's HB Mean Score

In [53]:
fig = go.Figure(data=[go.Histogram(x=HB_data['HB_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of HB mean score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### There are more contenders with high mean scores, 9 atheletes above 14, but only 2 above 14.4. 

### Distribution of Individual's HB SD

In [54]:
fig = go.Figure(data=[go.Histogram(x=HB_data['HB_std'],nbinsx=25)])
fig.update_layout(
    title='Histogram of HB SD',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### Performance on HB is generally more stable with a smaller SD, but difference between atheletes is quite high. It's easier to distinguish stable performers with others.

In [55]:
HB_sd_of_sd=HB_data['HB_std'].std()
HB_mean_sd=HB_data['HB_std'].mean()
print(HB_mean_sd.round(2),round(HB_sd_of_sd,2))

0.76 0.5


### Distribution of HB D_Score

In [56]:
fig = go.Figure(data=[go.Histogram(x=HB_data['HB_D_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of HB D_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### There is smaller difference in D_score among top-performers. Quite a few atheletes are able to achieve a high D_score above 5.9 (8 atheletes).

### Distribution of HB E_Score

In [57]:
fig = go.Figure(data=[go.Histogram(x=HB_data['HB_E_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of HB E_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### For each score, the best atheletes are distinguished from others, with 8 atheletes above 8.4. Below that, it is all clustered together.
### Conclusion
### For HB, USA has one outstanding athelete Brody Malone, who has some hope in winning the medal but there are still one definitely better than him. HB as an appratus disinguishes atheletes further, creating a gap between mediocre atheletes and the best. D_score seems to matter less in HB, and it's important to achieve a high E_score that sets atheletes apart.
## Men's Parallel Bars

### Top 15 Athletes in PB by mean_score (3+ Apperances)

In [84]:
PB_data = men_data[men_data['PB_count']>3]
PB_data=PB_data[['name','Country','PB_mean_score','PB_std','PB_count','PB_D_mean_score','PB_D_std','PB_E_mean_score','PB_E_std']]
PB_data.sort_values(by='PB_mean_score', ascending=False).head(10)

name Country  PB_mean_score  PB_std  PB_count  \
618     JINGYUAN ZOU     CHN          15.92    0.16       7.0   
35      LUKAS DAUSER     GER          15.15    0.57      10.0   
6        CARLOS YULO     PHI          15.08    0.29      14.0   
0       BOHENG ZHANG     CHN          15.02    0.53       8.0   
15      ILLIA KOVTUN     UKR          14.98    0.29      30.0   
533   KAITO SUGIMOTO     JPN          14.95    0.13       5.0   
397       DEHANG YIN     CHN          14.88    0.17       4.0   
2    WATARU TANIGAWA     JPN          14.86    0.07       4.0   
3           CONG SHI     CHN          14.84    0.16       5.0   
626  CURRAN PHILLIPS     USA          14.74    0.78      15.0   

     PB_D_mean_score  PB_D_std  PB_E_mean_score  PB_E_std  
618             6.61      0.20             9.30      0.11  
35              6.54      0.13             8.61      0.44  
6               6.34      0.11             8.73      0.27  
0               6.31      0.25             8.71      0.40  
15              6.40      0.17             8.58      0.25  
533             6.30      0.00             8.65      0.13  
397             6.40      0.00             8.48      0.17  
2               6.30      0.00             8.56      0.07  
3               6.18      0.16             8.66      0.08  
626             6.72      0.19             8.02      0.65

#### The best USA athelete in parallel bars is Curran Phillips, ranking 7th in the appratus. He is far behind the best in the apparatus, Jingyuan Zou, who has a much higher score on average and performs more consistently. However, Curren Phillips has the highest mean D_score among top 15, but has a lower E_score, indicating that he would take risks and attempt harder routines.
### Distribution of Men's PB Mean Score

In [59]:
fig = go.Figure(data=[go.Histogram(x=PB_data['PB_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of PB mean score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### One athelete sets himself apart from all others. In the second tier, there are 9 atheletes in the range of 14.8-15.2, which is quite a lot of top-performers. 14.2 is another cutoff point for many atheletes.
### Distribution of PB SD

In [60]:
filter_data_PB = PB_data[PB_data['PB_std']<3]
fig = go.Figure(data=[go.Histogram(x=filter_data_PB['PB_std'],nbinsx=30)])
fig.update_layout(
    title='Histogram of PB SD',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### Many atheletes are able to perform stably in PB, with 9 having a less than 0.2 SD and 19 lower than 0.3. It's hard to set atheletes apart by consistency.

In [61]:
#Get rid of outliers
filter_data_PB = PB_data[PB_data['PB_std']<2]
PB_sd_of_sd=filter_data_PB['PB_std'].std()
PB_mean_sd=filter_data_PB['PB_std'].mean()
print(PB_mean_sd.round(2),round(PB_sd_of_sd,2))

0.64 0.31


#### PB has a low mean of sd, and also a low sd of sd, meaning that atheletes are generally more consistent in this apparatus. Consistency is not what sets atheletes apart in this apparatus.
### Distribution of PB D_Score

In [62]:
fig = go.Figure(data=[go.Histogram(x=PB_data['PB_D_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of PB D_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### PB has a lot of atheletes achieving above 6, and 8 atheletes are able to score above 6.4. It means that D_score isn't what sets elite atheletes apart from each other.
### Distribution of PB's E_Score

In [63]:
fig = go.Figure(data=[go.Histogram(x=PB_data['PB_E_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of PB E_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### Only very few atheletes have high E_Scores, with 3 atheletes above 8.8, and those 3 are set apart as well. It shows that E_Score is the key difference between top-performing atheletes.
### Conclusion
### PB has an absolute leading athelete from China, and the best USA athelete is ranked 7th and has no chance winning the appratus medal. Atheletes are generally more stable in this apparatus, and many of them achieve high D_scores. What really matters in this apparatus is achieving a high E_score to be distinguished apart.
## Men's Pommel Horse
### Top 15 Athletes in PB by mean_score (3+ Apperances)

In [85]:
PH_data = men_data[men_data['PH_count']>3]
PH_data=PH_data[['name','Country','PH_mean_score','PH_std','PH_count','PH_D_mean_score','PH_D_std','PH_E_mean_score','PH_E_std']]
PH_data.sort_values(by='PH_mean_score', ascending=False).head(10)

name Country  PH_mean_score  PH_std  PH_count  \
837   MC RHYS CLENAGHAN     IRL          15.14    0.16       4.0   
628       RYOTA TSUMURA     JPN          14.88    0.43       6.0   
840    RHYS MCCLENAGHAN     IRL          14.71    0.38      16.0   
839    NARIMAN KURBANOV     KAZ          14.71    0.69      31.0   
48         CHIH KAI LEE     TPE          14.49    0.88      15.0   
844  HARUTYUN MERDINYAN     ARM          14.42    0.48      15.0   
1           KAZUMA KAYA     JPN          14.37    0.21       7.0   
453        YU JAN SHIAO     TPE          14.33    0.62      18.0   
847  STEPHEN NEDOROSCIK     USA          14.30    0.69      12.0   
397          DEHANG YIN     CHN          14.29    0.35       6.0   

     PH_D_mean_score  PH_D_std  PH_E_mean_score  PH_E_std  
837             6.38      0.05             8.77      0.19  
628             6.53      0.05             8.35      0.41  
840             6.31      0.17             8.40      0.39  
839             6.26      0.29             8.46      0.62  
48              6.28      0.27             8.21      0.76  
844             5.99      0.11             8.43      0.38  
1               6.31      0.11             8.05      0.17  
453             5.93      0.25             8.41      0.52  
847             6.36      0.43             7.94      0.52  
397             6.32      0.12             7.97      0.33

#### The best USA athelete in PH is Stephen Nedoroscik, ranking 5th among all. He is 0.58 points behind leading Ryota Tsumura, who has a higher D_score, E_score and better consistency. Stephen has one of the worst consistency among top 15 atheletes, but he has a relatively high D_score.
### Distribution of Men's PH Mean Score

In [65]:
fig = go.Figure(data=[go.Histogram(x=PH_data['PH_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of PH mean score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### 2 best atheletes have clear cut advantage in PH, and multiple top performers follow in the range of 14-14.4. The distribution is more widespread, showing that atheletes are generally more distinguished in PH.
### Distribution of Men's PH SD

In [66]:
fig = go.Figure(data=[go.Histogram(x=PH_data['PH_std'],nbinsx=30)])
fig.update_layout(
    title='Histogram of PH SD',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### For Men's Pommel Horse, there are few atheletes with low SD, meaning that the performances are generally inconsistent. 

In [67]:
#Get rid of outliers
filter_data_PH = PH_data[PH_data['PH_std']<2]
PH_sd_of_sd=filter_data_PH['PH_std'].std()
PH_mean_sd=filter_data_PH['PH_std'].mean()
print(PH_mean_sd.round(2),round(PH_sd_of_sd,2))

0.81 0.33


#### The mean of SD is high, meaning that generally atheletes' performances in PH is inconsistent.
### Distribution of PH D_Score

In [68]:
fig = go.Figure(data=[go.Histogram(x=PH_data['PH_D_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of PH D_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### Many atheletes are able to achieve a high D_score in pommel horse, meaning that it's hard to standout with D_Score.
### Distribution of PH E_Score

In [69]:
fig = go.Figure(data=[go.Histogram(x=PH_data['PH_E_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of PH E_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### Much fewer people are able to achieve a E_score higher than 8.2, meaning that E_score is what truly distinguishes good atheletes from the best.
### Conclusion
### For Pommel Horse, atheletes' performances are less consistent as their SD is large. D_score are less distinguishing than E_Score for top level atheletes. The best on USA team is Stephen Nedoroscik, who rank 4th among all and have little chance of winning gold. (but some chances in winning medal)
## Men's Still Ring
### Top 15 Athletes in SR by mean_score (3+ Apperances)

In [86]:
SR_data = men_data[men_data['SR_count']>3]
SR_data=SR_data[['name','Country','SR_mean_score','SR_std','SR_count','SR_D_mean_score','SR_D_std','SR_E_mean_score','SR_E_std']]
SR_data.sort_values(by='SR_mean_score', ascending=False).head(10)

name Country  SR_mean_score  SR_std  SR_count  \
635                YANG LIU     CHN          15.23    0.14       5.0   
13               XINGYU LAN     CHN          15.11    0.29       9.0   
618            JINGYUAN ZOU     CHN          14.80    0.32       7.0   
838  ELEFTHERIOS PETROUNIAS     GRE          14.80    0.39       9.0   
17                ADEM ASIL     TUR          14.78    0.27      27.0   
541                 HAO YOU     CHN          14.73    0.12       6.0   
650       SALVATORE MARESCA     ITA          14.71    0.08       6.0   
841          VAHAGN DAVTYAN     ARM          14.61    0.14      18.0   
552           IBRAHIM COLAK     TUR          14.60    0.34      15.0   
629           MARCO LODADIO     ITA          14.60    0.07       4.0   

     SR_D_mean_score  SR_D_std  SR_E_mean_score  SR_E_std  
635             6.40      0.00             8.83      0.14  
13              6.38      0.04             8.78      0.17  
618             6.27      0.13             8.53      0.23  
838             6.26      0.05             8.55      0.38  
17              6.29      0.03             8.48      0.26  
541             6.57      0.10             8.16      0.16  
650             6.20      0.00             8.51      0.08  
841             5.97      0.05             8.65      0.16  
552             6.09      0.06             8.50      0.36  
629             6.30      0.00             8.30      0.07

#### The best USA athelete in Still Rings is Donnell Whittenburg, who scores on average 14.54 with a 0.3 SD, and 15 appearances. He ranks 11th among all atheletes, 0.69 points behind the 1st ranked Yang Liu. He has a low mean D_score of 5.99 compared to other atheletes on the same level, while his E_score is on a higher level.
### Distribution of Men's SR Mean Score

In [71]:
fig = go.Figure(data=[go.Histogram(x=SR_data['SR_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of SR mean score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### It can be observed that the gap between elite atheletes and others is rather large in SR. 9 Atheletes in total could score over 14.6 on average. Top 2 atheletes are also score distinctly above others. 
### Distribution of Men's SR SD

In [72]:
fig = go.Figure(data=[go.Histogram(x=SR_data['SR_std'],nbinsx=25)])
fig.update_layout(
    title='Histogram of SR SD',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### This graph shows that many atheletes are able to perform stably in SR, with 12 people having a SD lower than 0.14.

In [73]:
#Get rid of outliers
filter_data_SR = SR_data[SR_data['SR_std']<2]
SR_sd_of_sd=filter_data_SR['SR_std'].std()
SR_mean_sd=filter_data_SR['SR_std'].mean()
print(SR_mean_sd.round(2),round(SR_sd_of_sd,2))

0.39 0.21


#### Both the mean of SD and SD of SD are very low in SR, showing that this apparatus allows consistent performance in general. 
### Distribution of SR D_Score

In [74]:
fig = go.Figure(data=[go.Histogram(x=SR_data['SR_D_mean_score'],nbinsx=250)])
fig.update_layout(
    title='Histogram of SR D_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### D_score is not very distinguishing in SR, as 9 atheletes are able to achieve an average D_Score in the top 2 tiers (6.2-6.6).
### Distribution of SR E_Score

In [75]:
fig = go.Figure(data=[go.Histogram(x=SR_data['SR_E_mean_score'],nbinsx=250)])
fig.update_layout(
    title='Histogram of SR E_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### Among the top-performing atheletes, E_score is the distinction. The highest few bins all have only 1-2 atheletes in each bin. This is where elite atheletes are distinguished in SR.
### Conclusion
### SR has some leading atheletes that are far above average ones, but unfortunately none of them are USA atheletes. There is little hope for USA winning apparatus medal in SR. This apparatus is generally consistent in atheletes' performances with lower SD in general, and E_score is what sets atheletes apart from each other.
## Men's Vault
### Top 15 Athletes in VT by mean_score (5+ Apperances)

In [87]:
VT_data = men_data[men_data['VT_count']>5]
VT_data=VT_data[['name','Country','VT_mean_score','VT_std','VT_count','VT_D_mean_score','VT_D_std','VT_E_mean_score','VT_E_std']]
VT_data.sort_values(by='VT_mean_score', ascending=False).head(10)

name Country  VT_mean_score  VT_std  VT_count  \
26          ARTUR DAVTYAN     ARM          14.89    0.36      50.0   
187       NAZAR CHEPURNYI     UKR          14.79    0.16      30.0   
27            JAKE JARMAN     GBR          14.76    0.50      42.0   
6             CARLOS YULO     PHI          14.69    0.40      25.0   
19             KHOI YOUNG     USA          14.69    0.30      15.0   
633        IGOR RADIVILOV     UKR          14.68    0.37      33.0   
2         WATARU TANIGAWA     JPN          14.65    0.44      11.0   
4         DAIKI HASHIMOTO     JPN          14.64    0.40       7.0   
747         KAZUKI MINAMI     JPN          14.58    0.34      10.0   
425  GIARNNI REGINI MORAN     GBR          14.56    0.53      12.0   

     VT_D_mean_score  VT_D_std  VT_E_mean_score  VT_E_std  
26              5.58      0.13             9.34      0.30  
187             5.60      0.00             9.20      0.16  
27              5.76      0.22             9.03      0.45  
6               5.63      0.23             9.07      0.36  
19              5.47      0.21             9.26      0.23  
633             5.60      0.00             9.09      0.35  
2               5.64      0.33             9.02      0.21  
4               5.60      0.00             9.13      0.28  
747             5.48      0.19             9.10      0.37  
425             5.47      0.20             9.10      0.39

#### Due to the special rule for VT, here only atheletes with 5+ appearances are considered. The best USA athelete in VT is Khoi Young, who scores on average 14.76, ranking 3rd among all. He is only 0.13 points behind the leading scorer Artur Davtyan from Armenia. Khoi is a medal contender in this apparatus.
### Distribution of Men's VT mean score

In [77]:
fig = go.Figure(data=[go.Histogram(x=VT_data['VT_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of VT mean score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### There are a handful of elite atheletes in VT, (8 scoring above 14.6), so there is tight competition on the top.
### Distribution of Men's VT SD

In [78]:
fig = go.Figure(data=[go.Histogram(x=VT_data['VT_std'],nbinsx=25)])
fig.update_layout(
    title='Histogram of VT SD',
    xaxis_title='Score',
    yaxis_title='Count'
)

In [79]:
#Get rid of outliers
filter_data_VT = VT_data[VT_data['VT_std']<2]
VT_sd_of_sd=filter_data_VT['VT_std'].std()
VT_mean_sd=filter_data_VT['VT_std'].mean()
print(VT_mean_sd.round(2),round(VT_sd_of_sd,2))

0.46 0.21


In [80]:
fig = go.Figure(data=[go.Histogram(x=VT_data['VT_D_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of VT D_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### D_score in men's VT can set atheletes apart on a higher level.

In [81]:
fig = go.Figure(data=[go.Histogram(x=VT_data['VT_E_mean_score'],nbinsx=25)])
fig.update_layout(
    title='Histogram of VT E_score',
    xaxis_title='Score',
    yaxis_title='Count'
)

#### E_score is clustered on the higher end, so it's not very distinguishing for elite atheletes.
### Conclusion
### In VT, USA has one 3rd athelete that can compete for medal. D_score is more important than E_score in VT, and performance is generally stable.